panther/db/connections.py

```diff
@@ -141,5 +148,56 @@ def create_connection_for_websocket(self) -> _Redis:
         return self.websocket_connection
 
 
+class ValkeyConnection(Singleton, _Valkey):
+    def __init__(
+            self,
+            init: bool = False,
+            host: str = 'localhost',
+            port: int = 6379,
+            db: int = 0,
+            websocket_db: int = 0,
+            **kwargs
+    ):
+        if init:
+            self.host = host
+            self.port = port
+            self.db = db
+            self.websocket_db = websocket_db
+            self.kwargs = kwargs
+
+            super().__init__(host=host, port=port, db=db, **kwargs)
+            self.sync_ping()
+
+    @property
+    def is_connected(self):
+        try:
+            self.sync_ping()
+        except valkey_exceptions.ConnectionError:
+            return False
+        else:
+            return True
+
+    def sync_ping(self):
+        from valkey import Valkey
+
+        Valkey(host=self.host, port=self.port, socket_timeout=3, **self.kwargs).ping()
+
+    async def execute_command(self, *args, **options):
+        if self.is_connected:
+            return await super().execute_command(*args, **options)
+        msg = '`VALKEY` is not found in `configs`'
+        raise ValueError(msg)
+
+    def create_connection_for_websocket(self) -> _Redis:
+        if not hasattr(self, 'websocket_connection'):
+            self.websocket_connection = _Redis(
+                host=self.host,
+                port=self.port,
+                db=self.websocket_db,
+                **self.kwargs
+            )
+        return self.websocket_connection
+
 db: DatabaseConnection = DatabaseConnection()
 redis: RedisConnection = RedisConnection()
+valkey: ValkeyConnection = ValkeyConnection()
```

Comment:
As we have so many places that use `redis` and we have to `or` them with `valkey`, I think it would be better to have a variable named `cache` which stores the correct value. 
And if we go with this new style, we have to have a new abstract class for the `redis` and `valkey` to save the structure